allennlp/data/dataset_readers/snli.py

```diff
@@ -0,0 +1,122 @@
+from typing import Dict, Optional
```

Comment:
Was this just copied over from `allennlp-models`?

Reply:
yes, it's used in the test, but the environment in which the test is run doesn't have `allennlp-models`.

Reply:
It's fine to have some tests in `allennlp-models` for this code. For instance, we have some checklist tests there, because that's where the models are.

Reply:
I added the test under `allennlp-models`.

Reply:
We can remove this file now, right?